### PR TITLE
feat(Deps/Boost): add boost support

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Start support `boost`

## Issues Addressed:
- Part #3451
- Part #4302
- Part #3739

Better after #5611

### CODE REVIEW

File: .github/workflows/windows_build.yml [hunk 1]
```diff
@@ -12,11 +12,13 @@ jobs:
       fail-fast: false
     runs-on: windows-2019
     name: windows-2019-MSVC16
+    env:
+      BOOST_ROOT: C:\local\boost_1_74_0
```
Comment (FrancescoBorzi) [inline .github/workflows/windows_build.yml:16]: any reasons not to use the latest version? currently 1.76 (released about 1 month ago)

File: apps/installer/includes/os_configs/ubuntu.sh [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-sudo apt-get update -y
+# Added repo for newest lib
+sudo add-apt-repository -y ppa:mhier/libboost-latest
+sudo apt update
 
 # shared deps
 sudo apt-get -y install make cmake clang curl unzip libmysqlclient-dev libace-dev ccache google-perftools
 
+# Insstall boost 1.73 from ppa:mhier/libboost-latest for all os versions
+sudo apt-get -y install libboost1.73-dev
```
Comment (FrancescoBorzi) [inline apps/installer/includes/os_configs/ubuntu.sh:16]: same as above, why using an older version?

File: .github/workflows/windows_build.yml (hunk 1 above)
Comment (Winfidonarleyan) [reply .github/workflows/windows_build.yml:16]: https://community.chocolatey.org/packages/boost-msvc-14.2

File: apps/installer/includes/os_configs/ubuntu.sh (hunk 2 above)
Comment (Winfidonarleyan) [reply apps/installer/includes/os_configs/ubuntu.sh:16]: 1.74 verison maximum

File: apps/installer/includes/os_configs/ubuntu.sh [hunk 3]
```diff
@@ -12,8 +12,8 @@ sudo apt update
 # shared deps
 sudo apt-get -y install make cmake clang curl unzip libmysqlclient-dev libace-dev ccache google-perftools
 
-# Insstall boost 1.73 from ppa:mhier/libboost-latest for all os versions
-sudo apt-get -y install libboost1.73-dev
+# Insstall boost 1.74 from ppa:mhier/libboost-latest for all os versions
+sudo apt-get -y install libboost1.74-dev
```
Comment (FrancescoBorzi) [inline apps/installer/includes/os_configs/ubuntu.sh:16]: better :)

Comment (Winfidonarleyan) [reply apps/installer/includes/os_configs/ubuntu.sh:16]: okay okay :D
I install `1.76` now
`CMake` support max `1.75` in version `20.2`

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**